addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -169,6 +178,10 @@ def _compute_is_pinned(self):
         'CHECK((partner_id IS NOT NULL AND guest_id IS NULL) OR (partner_id IS NULL AND guest_id IS NOT NULL))',
         'A channel member must be a partner or a guest.',
     )
+    _channel_owner_unique = models.UniqueIndex(
+        "(channel_id, channel_role) WHERE channel_role = 'owner'",
+        "A channel can only have maximum one owner.",
+    )
```

Comment:
Shouldn't we also have a constraint to disallow `channel_role` in non-supported channel type?